src/model/priors/single_cell_bulk.rs

```diff
@@ -0,0 +1,578 @@
+        let prob = LogProb::ln_sum_exp(&af_single.iter().map(|&af_single| {
+            let p_single =
+                    LogProb::ln_sum_exp(&k_single.clone().map(|k_s| { // sum up all possible discrete single cell allele frequencies with current number of observations
+                        let af_single_distorted = AlleleFreq(k_s as f64/n_obs_s as f64);
+                        likelihood_single_distorted(af_single_distorted, None) + self.prob_rho(&af_single, &n_obs_s, &k_s)
+                    }).collect_vec());
+            let prob = p_bulk + p_single;
+
+            prob
+        }).collect_vec());
+
+        prob
+    }
+
+    fn marginal_prob<L, O>(
+        &self,
+        likelihood_single_distorted: &L,
+        likelihood_bulk: &O,
+        variant: Variant,
+        n_obs_single: usize,
+        n_obs_bulk: usize
+    ) -> LogProb where
+        L: Fn(AlleleFreq, Option<AlleleFreq>) -> LogProb,
+        O: Fn(AlleleFreq, Option<AlleleFreq>) -> LogProb
+    {
+        self.joint_prob(
+            self.allele_freqs().0,
+            self.allele_freqs().1,
+            likelihood_single_distorted,
+            likelihood_bulk,
+            variant,
+            n_obs_single,
+            n_obs_bulk
+        )
+    }
+
+    fn map<L, O>(
+        &self,
+        likelihood_single_distorted: &L,
+        likelihood_bulk: &O,
+        _: Variant,
+        n_obs_single: usize,
+        n_obs_bulk: usize
+    ) -> (AlleleFreq, AlleleFreq) where
+        L: Fn(AlleleFreq, Option<AlleleFreq>) -> LogProb,
+        O: Fn(AlleleFreq, Option<AlleleFreq>) -> LogProb
+    {
+        // cap the use of the single cell amplification bias model at a coverage of 100,
+        // as the Lodato et al. model was fit with coverages capped at 60 and starts behaving
+        // weirdly above 100
+        let n_obs_s = if n_obs_single > 100 {
+            100
+        } else {
+            n_obs_single
+        };
+        let k_single = 0..n_obs_s + 1;
+        let (_, map_single) = self.allele_freqs().0.iter().minmax_by_key(
+            |&&af_single| {
+                let p_single =
+                    LogProb::ln_sum_exp(&k_single.clone().map(|k_s| { // sum up all possible discrete single cell allele frequencies with current number of observations
+                        let af_single_distorted = AlleleFreq(k_s as f64/n_obs_s as f64);
+                        likelihood_single_distorted(af_single_distorted, None) + self.prob_rho(&af_single, &n_obs_s, &k_s)
+                    }).collect_vec());
+                NotNaN::new(*p_single).expect("posterior probability is NaN")
+            }
+        ).into_option().expect("prior has empty allele frequency spectrum");
+
+        let k_bulk = 0..n_obs_bulk + 1;
+        let afs_bulk = k_bulk.map(|k| AlleleFreq(k as f64/n_obs_bulk as f64));
+        let (_, map_bulk) = afs_bulk.minmax_by_key(
+            |af_bulk| {
+                let p_bulk = likelihood_bulk(*af_bulk, None);
```

Comment:
Oops, that one ... got lost ...

Well, as the prior is currently flatly set to 1, I guess I just completely forget about it. Will of course add it back in, also for the bulk in fn joint_prob().